### PR TITLE
Add doc for recent target extension features

### PR DESCRIPTION
**Pull request by sklam**
- Add docs on compiler flag inheritance

### CODE REVIEW

File: docs/source/developer/target_extension.rst [hunk 1]
```diff
@@ -0,0 +1,62 @@
+=========================
+Notes on Target Extension
+=========================
```
Comment (stuartarchibald) [inline docs/source/developer/target_extension.rst:3]:
```suggestion
==========================
Notes on Target Extensions
==========================
```

File: docs/source/developer/target_extension.rst [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+.. warning:: All features and API described in this page are in-development and
+             may change without deprecation notice.
```
Comment (stuartarchibald) [inline docs/source/developer/target_extension.rst:6]:
```suggestion
.. warning:: All features and APIs described in this page are in-development and
             may change at any time without deprecation notices being issued.
```

File: docs/source/developer/target_extension.rst [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+
+Inheriting compiler flags from the caller
+=========================================
+
+Compiler flags, i.e. options such as ``fastmath``, ``nrt`` in
+``@jit(nrt=True, fastmath=True))`` are specified per-function but their
+effects are not well-defined---some flags affect the entire callgraph, some
+flags affect only the current function. Sometimes, it is necessary for callees
```
Comment (stuartarchibald) [inline docs/source/developer/target_extension.rst:15]:
```suggestion
flags affect only the current function. Sometimes it is necessary for callees
```

File: docs/source/developer/target_extension.rst [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+to inherit flags from the caller; for example the ``fastmath`` flag should be
+infectious.
+
+To address the problem, the followings are needed:
```
Comment (stuartarchibald) [inline docs/source/developer/target_extension.rst:19]:
```suggestion
To address the problem, the following are needed:
```

File: docs/source/developer/target_extension.rst [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+
+1. Better define the semantic for compiler flags. Preferably, all flags should
```
Comment (stuartarchibald) [inline docs/source/developer/target_extension.rst:21]:
```suggestion
1. Better definitions for the semantics of compiler flags. Preferably, all flags should
```

**Review by stuartarchibald — COMMENTED** [review]
Thanks for adding these docs, a few minor things to look at else looks good.

**Comment by sklam** [comment]
I was planning to add more documentation but since this is reviewed. I will keep it short and add more in a new PR.

**Review by stuartarchibald — APPROVED** [review]
Thanks for patch and fixes. I've added dealing with the "TODO"s to https://github.com/numba/numba/issues/6908

**Comment by stuartarchibald** [comment]
PR doc build is here: https://numba--6877.org.readthedocs.build/en/6877/developer/target_extension.html

**Pull request merged**